composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -16,18 +16,20 @@
 from composer.core.types import Batch
 from composer.loggers import Logger
 from composer.models import HuggingFaceModel
-from composer.utils import ensure_tuple
+from composer.utils import dist, ensure_tuple
 
 __all__ = ['SeqLengthWarmup', 'set_batch_sequence_length']
 
 
-def set_batch_sequence_length(batch: Dict[str, torch.Tensor], curr_seq_len: int, truncate: bool = True) -> Batch:
+def set_batch_sequence_length(
+    batch: Dict[str, torch.Tensor],
+    curr_seq_len: int,
+    preserve_end_of_sequence: bool = False,
+) -> Batch:
     """Set the sequence length of a batch.
 
     Changes the sequence length of all tensors in the provided dictionary
-    to ``curr_seq_len`` by either truncating the tensors (``truncate=True``)
-    or reshaping the tensors to create new examples from the extra tokens
-    (``truncate=False``).
+    to ``curr_seq_len`` by truncating the sequence tensors.
```

Comment:
Suggest the new option be explained in the docstrings perhaps with some example sequences?

Reply:
Expanded the docstrings to have such an example